website/src/docs/reference/schema.md

```diff
@@ -0,0 +1,836 @@
+---
+title: Schema Reference
+description: Complete reference for the Taskfile schema based on the official JSON schema
+outline: deep
+---
+
+# Schema Reference
+
+This page documents all available properties and types for the Taskfile schema version 3, based on the [official JSON schema](https://taskfile.dev/schema.json).
+
+## Root Schema
+
+The root Taskfile schema defines the structure of your main `Taskfile.yml`.
+
+### `version`
+
+- **Type**: `string` or `number`
+- **Required**: Yes
+- **Valid values**: `"3"`, `3`, or any valid semver string
+- **Description**: Version of the Taskfile schema
+
+```yaml
+version: '3'
+```
+
+### `output`
+
+- **Type**: `string` or `object`
+- **Default**: `interleaved`
+- **Options**: `interleaved`, `group`, `prefixed`
+- **Description**: Controls how task output is displayed
+
+```yaml
+# Simple string format
+output: group
+
+# Advanced object format
+output:
+  group:
+    begin: "::group::{{.TASK}}"
+    end: "::endgroup::"
+    error_only: false
+```
+
+### `method`
+
+- **Type**: `string`
+- **Default**: `checksum`
+- **Options**: `checksum`, `timestamp`, `none`
+- **Description**: Default method for checking if tasks are up-to-date
+
+```yaml
+method: timestamp
+```
+
+### `includes`
+
+- **Type**: `map[string]Include`
```

Comment:
Would be great if this could be a link to the `Include` section like we have in the current docs. Same goes for other references (e.g. `Variable`, `Task`). Doesn't have to work the same way as our current docs, but some kind of quick way to get to the inner type would be nice.